catalog/transformer/catalog-transformer-csv-common/src/test/java/ddf/catalog/transformer/csv/common/CsvTransformerTest.java

```diff
@@ -0,0 +1,231 @@
+
+    String[] expectedHeaders = {"attribute1"};
+    validate(scanner, expectedHeaders);
+
+    String[] expectedValues = {"", "value1"};
+
+    for (int i = 0; i < METACARD_COUNT; i++) {
+      validate(scanner, expectedValues);
+    }
+
+    // final new line causes an extra "" value at end of file
+    assertThat(scanner.hasNext(), is(true));
+    assertThat(scanner.next(), is(""));
+    assertThat(scanner.hasNext(), is(false));
+  }
+
+  @Test
+  public void writeSearchResultsToCsvWithAliasMap() throws CatalogTransformerException {
+    List<AttributeDescriptor> requestedAttributes = new ArrayList<>();
+    requestedAttributes.add(buildAttributeDescriptor("attribute1", BasicTypes.STRING_TYPE));
+
+    Map<String, String> aliasMap = ImmutableMap.of("attribute1", "column1");
+
+    Appendable csvText =
+        CsvTransformer.writeMetacardsToCsv(metacardList, requestedAttributes, aliasMap);
+
+    Scanner scanner = new Scanner(csvText.toString());
+    scanner.useDelimiter(CSV_ITEM_SEPARATOR_REGEX);
+
+    String[] expectedHeaders = {"column1"};
+    validate(scanner, expectedHeaders);
+
+    String[] expectedValues = {"", "value1"};
+
+    for (int i = 0; i < METACARD_COUNT; i++) {
+      validate(scanner, expectedValues);
+    }
+
+    // final new line causes an extra "" value at end of file
+    assertThat(scanner.hasNext(), is(true));
+    assertThat(scanner.next(), is(""));
+    assertThat(scanner.hasNext(), is(false));
+  }
+
+  private Metacard buildMetacard() {
+    MetacardType metacardType = new MetacardTypeImpl("", new HashSet<>(ATTRIBUTE_DESCRIPTOR_LIST));
+    Metacard metacard = new MetacardImpl(metacardType);
+    for (Attribute a : metacardDataMap.values()) {
+      metacard.setAttribute(a);
+    }
+    return metacard;
+  }
+
+  private void buildMetacardList() {
+    metacardList.clear();
+    for (int i = 0; i < METACARD_COUNT; i++) {
+      metacardList.add(buildMetacard());
+    }
+  }
+
+  private void buildMetacardDataMap() {
+    for (ImmutableTriple entry : ATTRIBUTE_DATA) {
+      String attributeName = (String) entry.getLeft();
+      Serializable attributeValue = (Serializable) entry.getMiddle();
+      AttributeType attributeType = (AttributeType) entry.getRight();
+      Attribute attribute = new AttributeImpl(attributeName, attributeValue);
+      metacardDataMap.put(attributeName, attribute);
+      ATTRIBUTE_DESCRIPTOR_LIST.add(buildAttributeDescriptor(attributeName, attributeType));
+    }
+  }
+
+  private AttributeDescriptor buildAttributeDescriptor(String name, AttributeType type) {
+    AttributeDescriptor attributeDescriptor = mock(AttributeDescriptor.class);
+    when(attributeDescriptor.getName()).thenReturn(name);
+    when(attributeDescriptor.getType()).thenReturn(type);
+    return attributeDescriptor;
+  }
+
+  private void validate(Scanner scanner, String[] expectedValues) {
+    for (String expectedValue : expectedValues) {
+      assertThat(scanner.hasNext(), is(true));
```

Comment:
May want to add a failure message to this assertion that includes the expected value. Otherwise the failure message will just say that it was false when it was expected to be true and we'll have no idea which value it failed at.